std/haxe/macro/Expr.hx

```diff
@@ -169,154 +213,603 @@ enum Unop {
+
+	/**
+		Represents the structure to the type inheritance structure.
+	**/
 	TExtend( p : Array<TypePath>, fields : Array<Field> );
+
+	/**
+		Represents the optional type.
+	**/
 	TOptional( t : ComplexType );
 }
 
+/**
+	Represents a type path for the AST.
+**/
 typedef TypePath = {
+	/**
+		Represents the package of the type path. 
+	**/
 	var pack : Array<String>;
+
+	/**
+		The name of the type path.
+	**/
 	var name : String;
+
+	/**
+		Optional parameters of the type path.
+	**/
 	@:optional var params : Array<TypeParam>;
+
+	/**
+		
+	**/
 	@:optional var sub : Null<String>;
 }
 
+/**
+	Represents a type parameter for the AST.
+**/
 enum TypeParam {
+	/**
+		The type of the parameter.
+	**/
 	TPType( t : ComplexType );
+
+	/**
+		The expression node of the parameter.
+	**/
 	TPExpr( e : Expr );
 }
 
+/**
+	Represents a type parameter declaration for the AST.
+**/
 typedef TypeParamDecl = {
+	/**
+		The name of the type parameter declaration.
+	**/
 	var name : String;
+
+	/**
+		The constraints of the type parameter declaration.
+	**/
 	@:optional var constraints : Array<ComplexType>;
+
+	/**
+		The parameters of the type parameter declaration.
+	**/
 	@:optional var params : Array<TypeParamDecl>;
 }
 
+/**
+	Represents a function for the AST.
+**/
 typedef Function = {
+	/**
+		A list of function arguments.
+	**/
 	var args : Array<FunctionArg>;
+
+	/**
+		The return type of the function.
+	**/
 	var ret : Null<ComplexType>;
+
+	/**
+		The expression of the function body.
+	**/
 	var expr : Null<Expr>;
+
+	/**
+		An optional list of function parameter type declarations.
+	**/
 	@:optional var params : Array<TypeParamDecl>;
 }
 
+/**
+	Represents a function argument for the AST.
+**/
 typedef FunctionArg = {
+	/**
+		The name of the function argument.
+	**/
 	var name : String;
+
+	/**
+		Whether or not the function argument is optional.
+	**/
 	@:optional var opt : Bool;
+
+	/**
+		The type of the function argument.
+	**/
 	var type : Null<ComplexType>;
+
+	/**
+		The optional value of the function argument.
+	**/
 	@:optional var value : Null<Expr>;
 }
 
+/**
+	Represents a metadata entry for the AST.
+**/
 typedef MetadataEntry = {
+	/**
+		The name of the metadata entry.
+	**/
 	name : String,
+
+	/**
+		The optional parameters of the metadata entry.
+	**/
 	?params : Array<Expr>,
+
+	/**
+		The position of the metadata entry.
+	**/
 	pos : Position
 }
 
+/**
+	Represents metadata for the AST.
+**/
 typedef Metadata = Array<MetadataEntry>;
 
+/**
+	Represents a field for the AST.
+**/
 typedef Field = {
+	/**
+		The name of the field.
+	**/
 	var name : String;
+
+	/**
+		The documentation of the field. If the field has no documentation, the 
+		value is `null`.
+	**/
 	@:optional var doc : Null<String>;
+
+	/**
+		The access modifiers of the field. By default fields have private access.
+		@see http://haxe.org/manual/class-field-access-modifier.html
+	**/
 	@:optional var access : Array<Access>;
+
+	/**
+		The type of the field.
```

Comment:
"type" is really overloaded, let's stick to "kind". I know it's a bit confusing...